### PR TITLE
ENH: (+minor DOC fix) pretty print IncompleteResultsError

### PR DESCRIPTION
**Pull request by yarikoptic**
Someone recently had asked me why we are not having "multiline" exception renderings... and I forgot what actually was the reason for us (may be if only for logging) why we leaned toward one liners.

I think the change is trivial enough to be submitted against maint.

    ENH: Use pprint.pformat for IncompleteResultsError str
    
    Those are are now more rare but when they come, they become hard to parse visually,
    see e.g. one in https://github.com/datalad/datalad/pull/4581#issuecomment-636239700 .
    With pprint they become much more human friendly/informative even though might occupy
    considerable amount of terminal output space.
    
    E.g. here is the exception from that issue pprinted
    
            ======================================================================
            ERROR: test.test
            ----------------------------------------------------------------------
            Traceback (most recent call last):
              File "/usr/lib/python3/dist-packages/nose/case.py", line 197, in runTest
                    self.test(*self.arg)
              File "/tmp/testds/test.py", line 5, in test
                    raise IncompleteResultsError(
            datalad.support.exceptions.IncompleteResultsError: None [{'action': 'create_sibling',
              'message': ('failed to run post-update hook under remote path %s (%s)',
                                      'C:\\Users\\appveyor\\AppData\\Local\\Temp\\1\\datalad_temp_3jb32evv',
                                      'CommandError: \'"cd '
                                      '""C:\\Users\\appveyor\\AppData\\Local\\Temp\\1\\datalad_temp_3jb32evv\\.git"" '
                                      '&& ( [ -x hooks/post-update ] && hooks/post-update || : )"\' '
                                      "failed with exitcode 1 [err: '':' is not recognized as an "
                                      'internal or external command,\r\n'
                                      "operable program or batch file.'] [cmd.py:run:930]"),
              'orig_request': 'C:\\Users\\appveyor\\AppData\\Local\\Temp\\1\\datalad_temp_hh9cpvut',
              'path': 'C:\\Users\\appveyor\\AppData\\Local\\Temp\\1\\datalad_temp_hh9cpvut',
              'raw_input': True,
              'refds': 'C:\\Users\\appveyor\\AppData\\Local\\Temp\\1\\datalad_temp_hh9cpvut',
              'status': 'error',
              'type': 'dataset'}]
    
    it has not become the prettiest thing, BUT it became much more approachable

### CODE REVIEW

**Comment by yarikoptic** [comment]
I know that it is very minor and not urgent etc, so feel free to demote from 0.13.0

**Comment by yarikoptic** [comment]
and only now mentioned that it is actually even against `maint`... could go in either I guess.  Just would appreciate feedback ;-)

File: datalad/support/exceptions.py [hunk 1]
```diff
@@ -330,7 +332,7 @@ def __init__(self, results=None, failed=None, msg=None):
     def __str__(self):
         super_str = super(IncompleteResultsError, self).__str__()
         return "{} {}" \
```
Comment (bpoldrack) [inline datalad/support/exceptions.py:334]: May be then also make it `"{}{}{}".format(super_str, os.linesep, pformat(self.failed))` at least?
Still separating both by a space seems a bit weird if it's not a one-line output anymore. In your example output I find the `None [ ...` completely unclear wrt what it's trying to tell me.

Comment (yarikoptic) [reply datalad/support/exceptions.py:334]: Re None - note that I am only changing formatting of failed, and that None is the absent message given by super. I am only trying to make exception message more human parsable, not absolve all of its sins ;-)

Comment (bpoldrack) [reply datalad/support/exceptions.py:334]: Yes, I'm aware. What I'm saying is, that if we generate mutliline output anyway, that it's weird to have that one separated by a space only, suggesting some connection to what follows in that line.

Comment (yarikoptic) [reply datalad/support/exceptions.py:334]: Yeap yeap - I am putting my pants on already and will go downstairs to my laptop to tune it all up. I agree, it would be useful to provide better context of we are to improve it, but then just a new line probably wouldn't suffice

**Comment by yarikoptic** [comment]
ok, pushed my tune up. See e1951ce3ab6e9b8f7b9aaea1b1482e4e95ce22b5 for an example of rendering etc

**Comment by bpoldrack** [comment]
Tune up LGTM. Fail to see why Travis didn't pick it up, though.
